src/Gui/SoFCColorBar.cpp

```diff
@@ -78,6 +80,32 @@ void SoFCColorBarBase::GLRenderBelowPath(SoGLRenderAction *  action)
     SoSeparator::GLRenderBelowPath(action);
 }
 
+float SoFCColorBarBase::getBoundingWidth(const SbVec2s& size)
+{
+    // These are the same camera settings for front nodes as defined in the 3d view
+    SoOrthographicCamera* cam = new SoOrthographicCamera;
+    cam->position = SbVec3f(0, 0, 5);
+    cam->height = 10;
+    cam->nearDistance = 0;
+    cam->farDistance = 10;
+
+    SoGroup* group = new SoGroup();
+    group->ref();
+    group->addChild(cam);
+    group->addChild(this);
+
+    SbViewportRegion vpr(size);
+    SoGetBoundingBoxAction bbact(vpr);
+    bbact.apply(group);
+    SbBox3f box = bbact.getBoundingBox();
+    SbVec3f minPt, maxPt;
+    box.getBounds(minPt, maxPt);
+    group->unref();
+
+    float boxWidth = maxPt[0] - minPt[0];
+    return boxWidth + 0.2f;
```

Comment:
why the 0.2f extra? I thought the routine should return the actual width and the final adjustment is later done in `SoFCColorGradient::setViewportSize`

Reply:
When I omit the 0.2f extra it looks better here since there is then less whitespace until the right toolbar begins. I will try out tomorrow on all PCs I can access and report back.

Reply:
> why the 0.2f extra? I thought the routine should return the actual width and the final adjustment is later done in SoFCColorGradient::setViewportSize

This is a small offset to avoid that the text is too close to the right border. It doesn't need to be 0.2 it can also be 0.1 or 0.05 or so.